### ISSUE TITLE
AddDistributedRedisCache() working with Sentinel, where it shouldn't be able to. 

### ISSUE THREAD

**Issue by bobekhj** [aspnet/AspNetCore#8442]
### TL;DR How is AddDistributedRedisCache() working with a sentinel if StackExchange.Redis does not have that capability?

My setup:
I have 3 Redis servers with setup like `Example 2: basic setup with three boxes` here: https://redis.io/topics/sentinel

In my Asp.Net Core 2.* applications I have this code in Startup:

```
services.AddDistributedRedisCache(options =>
                {
                    options.Configuration = "redis1,redis2,redis3";
                });
```

Where `redis1` is a master by default and `redis2` and `redis3` are both slaves.

Now, here is the test I do. I take `redis1` offline, and my redis configuration automatically changes, `redis2` becomes master. At this point, if my code cannot handle the sentinel, I would expect it to fail. I should query the sentinel for new master and change my configuration.
